src/main/java/org/example/springproject/controller/CategoryController.java

```diff
@@ -0,0 +1,73 @@
+package org.example.springproject.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.example.springproject.dto.book.BookDtoWithoutCategoryIds;
+import org.example.springproject.dto.category.CategoryDto;
+import org.example.springproject.dto.category.CreateCategoryRequestDto;
+import org.example.springproject.service.BookService;
+import org.example.springproject.service.CategoryService;
+import org.springdoc.core.annotations.ParameterObject;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```

Comment:
```suggestion
@RestController
@Validated
```

as you use `@Positive` for Long id